scripts/configurator/parser/cmake_var.py

```diff
@@ -1,9 +1,98 @@
-from typing import Dict, Any, Type, Mapping
+from collections import namedtuple
+from typing import Dict, Any, Optional, Type, Mapping, List
+import re
+from sys import stderr
+
+
+class metadata:
+   """By convention, first row is considered a user comment
+   and is not considered in the metadata parsing process.
+   Raises a ValueError if a keyword definition is repeated twice, or if an
+   invalid keyword is provided."""
+   keyword_expr = re.compile(r"^([A-Z_]+) (.*)")
+
+   def __init__(self, comment: List[str]) -> None:
+      kw_tuple = namedtuple("kw_tuple", ["parser", "setter", "getter"])
+      """ If we wanted to use typehinting on kw_tuple, we should derive a new
+      class from namedtuple and set arguments directly from there"""
+      self.depends_on: Optional[List[str]] = None
+      self.group: Optional[str] = None
+      self.realtype: Optional[str] = None
+      self.kw = {
+         "REALTYPE": kw_tuple(self.parse_realtype,
+                              self.get_setter("realtype"),
+                              self.get_getter("realtype")),
+         "DEPENDS_ON":
+            kw_tuple(self.parse_depends_on,
+                     self.get_setter("depends_on"),
+                     self.get_getter("depends_on")),
+         "GROUP":
+            kw_tuple(self.parse_group,
+                     self.get_setter("group"),
+                     self.get_getter("group")),
+      }
+
+      for i, line in enumerate(comment):
+         match = self.keyword_expr.match(line)
+         if not match:
+            continue
+
+         key = match.group(1)
+         if i == 0:
+            if key in self.kw:
+               msg = """WARNING: First line uses a reserved keyword: {}.
+               This will be ignored."""
+               stderr.write(msg.format(key))
+            continue
+
+         elif key not in self.kw:
+            msg = "{} is not a valid metadata keyword."
+            raise ValueError(msg.format(key))
+
+         elif self.kw[key].getter() is not None:
+            msg = "Key {} has already a value associated to itself."
+            raise ValueError(msg.format(key))
+
+         setter = self.kw[key].setter
+         parsed_value = self.kw[key].parser(match.group(2))
+         setter(parsed_value)
+         assert(self.kw[key].getter() is not None)
+
+   def get_setter(self, var_name: str):
+
+      def setter(val: List[str] | str) -> None:
```

Comment:
The type hints for `val` are not OK. This should really be `any`. Don't try to forcibly put a type even in places where the type could be _anything by design_. The point of type hints is to put a type where _there is_ type and it's always the same. When we want something to support any type, we shouldn't put the list of types that we currently support: that would be an anti-pattern. We're trying to take advantage, from time to time, of the dynamic nature of Python. In C we would have used a `void *` or a discriminated union because we had no better choice. In Java we would have just used `Object`.

Reply:
Yeah, that's right. it's already removed. It also caused the CI to fail, because i was using the new syntax specific to 3.10